keps/0004-add-testing-infrastructure.md

```diff
@@ -84,7 +84,41 @@ exists in real world situations.
 
 ### CICD
 
-Identify and implement a CICD toolkit that can be integrated into GitHub to validate all PRs.
+Use [CircleCI](https://circleci.com/docs/) and the [GitHub Plugin](https://github.com/marketplace/circleci/plan/MDIyOk1hcmtldHBsYWNlTGlzdGluZ1BsYW45MA==#pricing-and-setup).
+For OpenSource projects we will recieve 1,000 monthly build minutes.  With the test suite below, that should suffice as a baseline.
+
+#### Pull Requests
+
+All Pull Requests into master need to have the following checks pass.  These should be ordered in fastest to slowest to reduce the time spent when/if failures occur
+
+0) Check author has signed CLA
+1) `go fmt` does not change anything 
+2) `go lint` and `go vet` both pass
+3) All unit tests pass (with `-race` flag)
+4) Dockerfile builds (this requires all dependencies in the vendor folder)
+
+Perform the same set of tests after merging master into the branch.
```

Comment:
Is that necessary or would it add a level of redundancy? Would we trigger 0-4 with every new commit in a pull?

Reply:
I think it's a good practice if you don't require all PRs always be rebased (which is pain). So I support that 👍